source/opt/ccp_pass.h

```diff
@@ -67,6 +67,13 @@ class CCPPass : public MemPass {
   // otherwise.
   bool ReplaceValues();
 
+  // Marks |instr| as varying by registering a varying value for its result
+  // into the |values_| table. Returns SSAPropagator::kVarying.
+  SSAPropagator::PropStatus MarkInstructionVarying(ir::Instruction* instr);
+
+  // Returns true if |id| is a varying value.
```

Comment:
The parameter name seem slightly confusing. |id| is not a SSA id, but I think that would the intuition. Suggest changing it to |val|.

Reply:
Recommend renaming the method to IsVaryingValueId.
And the method comment should be 

// Returns true if |id| is the sentinel id indicating a varying value.  It is never a valid SSA id.


OR: delete the method and test against kVaryingSentinelValueId everywhere.  That is probably less readable, though.